lib/repl.js

```diff
@@ -338,7 +338,8 @@ function REPLServer(prompt,
           }
         }
       } catch (e) {
-        if (isExitCommand) {
+        const context = this.useGlobal ? global : this.context;
+        if (isExitCommand && !context.hasOwnProperty(trimCommand)) {
```

Comment:
this can be done before executing the code, right?

Reply:
You mean define the `context` out of the `catch` block ?

Reply:
No. the whole test condition. check this before executing vm.runInContext

Reply:
ok I got it and I have modified the code, thank u~